rfc/rfc-0007-resource-references.md

```diff
@@ -0,0 +1,42 @@
+# RFC-0007: Resource References
+
+**Authors:** Scott Andrews
+
+**Status:**
+
+**Pull Request URL:** [#1370](https://github.com/projectriff/riff/pull/1370)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** [RFC 0006: Guidance for child resource names](https://github.com/projectriff/riff/pull/1369)
+
+
+## Problem
+It's common for one resource to reference another. For example, a Deployer can reference a Function build to obtain the latest image, or a stream provider references the gateway Service it created. riff should have a common way to express relationships between Kubernetes resources.
+
+### Anti-Goals
+This RFC does not define how specific riff resources reference other resources. It is only providing guidance.
+
+## Solution
+The Kubernetes API project provides two mechanisms for referencing other resources: ObjectReference and LocalObjectReference.
+
+[ObjectReference](https://godoc.org/k8s.io/api/core/v1#ObjectReference) has full support for referencing any k8s resource, commonly by `apiVersion`, `kind`, `namespace`, and `name`.
+
+[LocalObjectReference](https://godoc.org/k8s.io/api/core/v1#LocalObjectReference) contains only a resource's `name`. The `apiVersion`, `kind` and `namespace` must be inferred.
+
+riff should favor using ObjectReferences when the kind or namespace may vary or is unknown. LocalObjectReferences should be used when only the name of the resource will vary. A LocalObjectReference may be upgraded to an ObjectReference if explicitness is desired, however, it imposes a higher burden on those using the resource. `nil` should replace the reference if the referent is not defined.
+
+A validating webhook (or equivalent) should validate ObjectReferences before accepting the resource. The validation may restrict which fields are required and/or prohibited (`name` is typically required, `revision` is typically prohibited). The `apiVersion` and `kind` may be left open or restricted to known values.
```

Comment:
> *may* validate ObjectReferences



https://github.com/knative/serving/pull/5744

Reply:
@dprotaso this isn't intended to be an assertion of whether or not the referenced resource exists, but whether the reference itself is semantically valid

Reply:
sounds good - I don't have an approve button so go nuts

Reply:
found it